float32 linear_vx
float32 linear_vy
float32 linear_vz